optimum/onnxruntime/modeling_decoder.py

```diff
@@ -318,11 +340,8 @@ def _from_pretrained(
                         revision=revision,
                     )
                 except FileNotFoundError as e:
-                    raise FileNotFoundError(
-                        "The parameter `use_cache=True` was passed to ORTModelDecoder.from_pretrained()"
-                        " but no ONNX file using past key values could be found in"
-                        f" {str(Path(model_id, subfolder))}, with the error:\n    {e}"
-                    )
+                    # Raise if the `decoder` doesn't contain both w/o and w/. past IR
+                    cache_error = e
```

Comment:
Where is it raised?

Reply:
We gonna check then if the decoder is a merged one here:

https://github.com/huggingface/optimum/blob/995a9767f67c693726ee11987a079f9aad5221c8/optimum/onnxruntime/modeling_decoder.py#L432

If `use_cache=True`, no decoder with past found, and the decoder is not a merged one, the error will be raised.

Here is not necessary, previously I raise later because I will try merging the decoder, here you can raise directly, if no decoder with past + use_cache is True, and the decoder is not a merged one you raise.